README.md

```diff
@@ -1,52 +1,205 @@
+
+Example:
+
+```yaml
+spec:
+  deployKubeProxy: false
+  kubeProxyConfig:
+   iptablesSyncPeriod: 30s
+   bindAddress: 0.0.0.0
+   proxyArguments:
+     iptables-min-sync-period: ["30s"]
+```
+
+# Using
+The operator looks for a CRD of type `NetworkConfig` with the name `default`. Whenever this object is created or changed, the operator will apply those changes to the cluster (if they are safe to do so).
+
+You can view and edit that object with
+```
+kubectl edit networkconfig default
+```
+
+and any changes will be automatically applied.
+
+## Unsafe changes
+Most network changes are unsafe to roll out to a production cluster. Therefore, the network operator will stop reconciling if it detects that an unsafe change has been requested. 
+
+### Safe changes to apply:
+It is safe to edit the following fields in `NetworkConfig.Spec`:
+* deployKubeProxy
+* all of kubeProxyConfig
+
+### Force-applying an unsafe change
+Administrators may wish to forcefully apply a disruptive change to a cluster that is not serving production traffic. To do this, first they should make the desired configuration change to the CRD. Then, delete the network operator's understanding of the state of the system:
+
+```
+kubectl -n openshift-cluster-network-operator delete configmap applied-default
+```
+
+Be warned: this is an unsafe operation! It may cause the entire cluster to lose connectivity or even be permanently broken. For example, changing the ServiceNetwork will cause existing services to be unreachable, as their ServiceIP won't be reassigned.
+
+# Development
+
+## Basic structure
+
+The network operator consists of a controller loop and a rendering system. The basic flow is:
+
+1. The controller loop detects a configuration change
+1. The configuration is preprocessed:
+    - validity is checked.
+    - unspecified values are defaulted or carried forward from the previous configuration.
+    - safety of the proposed change is checked.
+1. The configuration is rendered in to a set of kubernetes objects (e.g. DaemonSets).
+1. The desired objects are reconciled against the API server, being created or updated as necessary.
+1. The applied configuration is stored separately, for later comparison.
+
+### Filling defaults
+Because most of the operator's configuration parameters are not changeable, it is important that the applied configuration is stable across upgrades. This has two implications:
+
+**All defaults must be made explicit.**
+
+The network configuration is transformed internally in to a fully-expressed struct. All optional values must have their default set. For example, if the vxlanPort is not specified, the default of 4789 is chosen and applied to the OpenShiftSDNConfig.
+
+Making all defaults explicit makes it possible to prevent unsafe changes when a newer version of the operator changes a default value.
```

Comment:
This should be at the top of the doc and reflected in the examples. Very important point!

Reply:
no, it's an implementation detail